src/components/Comments.jsx

```diff
@@ -14,7 +17,8 @@ export function Comments({ reviewId }) {
       })
       .catch((e) => {
         console.error(e);
-        window.alert("ERROR: Could not contact the server, try again later");
+        setHasError(true);
+        setErrorMessage("Encountered an issue fetching comments, please try again later");
```

Comment:
You may find that you can use a single piece of state to manage the error here - for example, the empty string can double up as `hasError === false`